common/src/main/java/io/netty/util/ConstantPool.java

```diff
@@ -68,7 +68,7 @@ public T valueOf(String name) {
     private T getOrCreate(String name) {
         T constant = constants.get(name);
         if (constant == null) {
-            final T tempConstant = newConstant(nextId(), name);
+            final T tempConstant = newConstant(0, name);
```

Comment:
this is change in semantics. Inheriting classes may depend upon this id to be "uniquely" generated. perhaps we should add another protected method `protected abstract T newConstant(String name);` and use this new method instead (and `@Deprecate` the existing method)?